### PR TITLE
fix: require confirmation on attachment removal and when cancelling edit

### PR DESCRIPTION
**Pull request by sirtawast**
## Description :sparkles:

Add a couple of missing nags with `window.confirm()`. It's lacking the cool HDS modal styles but at least it's working 😏  

Also set default paper application date if it's untouched.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2875.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://tet-pr2875.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2875.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2875.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-admin-ui-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-employer-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://kesaseteli-handler-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-handler-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-youth-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/benefit/handler/src/components/applicationForm/formContent/attachmentsList/useAttachmentsList.ts [hunk 1]
```diff
@@ -14,13 +14,14 @@ type ExtendedComponentProps = {
   attachments: [];
   isRemoving: boolean;
   isUploading: boolean;
-  handleRemove: (attachmentId: string) => void;
+  handleRemove: (attachmentId: string, name: string) => void;
```
Comment (EmiliaMakelaVincit) [inline frontend/benefit/handler/src/components/applicationForm/formContent/attachmentsList/useAttachmentsList.ts:17]: The signature of the actual returned function didn't change; there's no name parameter there. The name is resolved inside that function by using the ID alone.

**Review by EmiliaMakelaVincit — APPROVED** [review]
One typing mishap aside, seems fine.

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2875.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://tet-pr2875.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2875.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-admin-ui-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://kesaseteli-handler-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-handler-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-youth-ui-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2875.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-youth-pr2875.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**